eZ/Publish/Core/Persistence/Legacy/Content/UrlWildcard/Gateway/DoctrineDatabase.php

```diff
@@ -159,6 +173,39 @@ public function loadUrlWildcardsData(int $offset = 0, int $limit = -1): array
         return $stmt->fetchAll(FetchMode::ASSOCIATIVE);
     }
 
+    public function find(Criterion $criterion, int $offset, int $limit, array $sortClauses = [], bool $doCount = true): array
+    {
+        $count = $doCount ? $this->doCount($criterion) : null;
+        if (!$doCount && $limit === 0) {
+            throw new RuntimeException('Invalid query. Cannot disable count and request 0 items at the same time');
+        }
+
+        if ($limit === 0 || ($count !== null && $count <= $offset)) {
+            return [
+                'count' => $count,
+                'rows' => [],
+            ];
+        }
+
+        $query = $this->buildLoadUrlWildcardDataQuery();
+        $query
+            ->where($this->criteriaConverter->convertCriteria($query, $criterion))
+            ->setMaxResults($limit > 0 ? $limit : PHP_INT_MAX)
```

Comment:
Fetching results without pagination should be forbidden.

Reply:
Why? If someone wants to get all URL wildcards in accordance with the criteria, without using the limit, we should not let him?

This code does not have to be used only for the UI

Reply:
> Why? If someone wants to get all URL wildcards in accordance with the criteria, without using the limit, we should not let him?

Memory constraints. If you really need process all results, then e.g. \Ibexa\Contracts\Core\Repository\Iterator\BatchIterator should be used.